perses/annihilation/ncmc_switching.py

```diff
@@ -538,52 +622,9 @@ def __init__(self, temperature=default_temperature, functions=None,
                                                platform=platform, write_ncmc_interval=write_ncmc_interval,
                                                integrator_type=integrator_type)
 
-    def compute_logP(self, system, positions, parameter=None):
-        """
-        Compute potential energy of the specified system object at the specified positions.
-
-        Constraints are applied before the energy is computed.
-
-        Parameters
-        ----------
-        system : simtk.openmm.System
-            The System object for which the potential energy is to be computed.
-        positions : simtk.unit.Quantity with dimension [natoms, 3] with units of distance.
-            Positions of the atoms for which energy is to be computed.
-
-        Returns
-        -------
-        potential : simtk.unit.Quantity with units of energy
-            The computed potential energy
-            
-        """
-        # Create dummy integrator.
-        integrator = openmm.VerletIntegrator(self.timestep)
-        # Set the constraint tolerance if specified.
-        if self.constraint_tolerance is not None:
-            integrator.setConstraintTolerance(self.constraint_tolerance)
-        # Create a context on the specified platform.
-        if self.platform is not None:
-            context = openmm.Context(system, integrator, self.platform)
-        else:
-            context = openmm.Context(system, integrator)
-        context.setPositions(positions)
-        context.applyConstraints(integrator.getConstraintTolerance())
-        if parameter is not None:
-            available_parameters = self._getAvailableParameters(system)
-            for parameter_name in available_parameters:
-                context.setParameter(parameter_name, parameter)
-        # Compute potential energy.
-        potential = context.getState(getEnergy=True).getPotentialEnergy()
-        # Clean up context and integrator.
-        del context, integrator
-        # Return potential energy.
-        return -self.beta * potential
-
-    def _computeAlchemicalCorrection(self, unmodified_old_system,
-                                     unmodified_new_system, alchemical_system,
-                                     initial_positions, alchemical_positions,
-                                     final_hybrid_positions, final_positions,
+    def _computeAlchemicalCorrection(self, integrator, context,
```

Comment:
This `_computeAlchemicalCorrection` is different from the other one on line 161. Can they be combined, or do they need to be distinct?